test/libsolidity/SolidityNameAndTypeResolution.cpp

```diff
@@ -2107,7 +2107,7 @@ BOOST_AUTO_TEST_CASE(array_with_nonconstant_length)
 			function f(uint a) public { uint8[a] x; }
 		}
 	)";
-	CHECK_ERROR(text, TypeError, "Identifier must be declared constant.");
+	CHECK_ERROR(text, TypeError, "Invalid array length, expected integer literal or constant expression.");
```

Comment:
We lose a little precision here, but I think the previous error message was not particularly helpful, because you actually cannot declare `a` constant.